clientapi/routing/directory.go

```diff
@@ -117,12 +117,14 @@ func SetLocalAlias(
 	// 1. The new method for checking for things matching an AS's namespace
 	// 2. Using an overall Regex object for all AS's just like we did for usernames
 	for _, appservice := range cfg.Derived.ApplicationServices {
-		if aliasNamespaces, ok := appservice.NamespaceMap["aliases"]; ok {
-			for _, namespace := range aliasNamespaces {
-				if namespace.Exclusive && namespace.RegexpObject.MatchString(alias) {
-					return util.JSONResponse{
-						Code: http.StatusBadRequest,
-						JSON: jsonerror.ASExclusive("Alias is reserved by an application service"),
+		if device.UserID != appservice.SenderLocalpart {
```

Comment:
May be worth leaving a comment here that the UserID of the appservice's device will be a localpart, instead of a full UserID.

Frankly, I wonder if this will be slightly confusing down the line and instead we should set a full ID with the local serverName here. https://github.com/matrix-org/dendrite/blob/bc382bba4623a1c7ae16535541f2f4b23708999c/clientapi/auth/auth.go#L121